pkgs/tools/networking/p2p/jesec-rtorrent/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, makeWrapper
+, jesec-libtorrent
+, cacert
+, curl
+, ncurses
+, xmlrpc_c
+, nlohmann_json
+, xmlRpcSupport ? true
+, jsonRpcSupport ? true
+}:
+let inherit (lib) optional optionalString;
+in
+stdenv.mkDerivation rec {
+  pname = "rtorrent";
+  version = "0.9.8-r10";
+
+  src = fetchFromGitHub {
+    owner = "jesec";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Ge5W1rLaneUA7LxnBuMO/jQlqLOUKFf3gaAAlRr/qeM=";
+  };
+
+  nativeBuildInputs = [ cmake makeWrapper ];
+  buildInputs = [ jesec-libtorrent curl ncurses ]
+    ++ optional xmlRpcSupport xmlrpc_c
+    ++ optional jsonRpcSupport nlohmann_json;
+
+  cmakeFlags = [
+    (optionalString (!xmlRpcSupport) "-DUSE_XMLRPC=NO")
+    (optionalString (!jsonRpcSupport) "-DUSE_JSONRPC=NO")
+  ];
+
+  postFixup = ''
+    mv $out/bin/rtorrent $out
+    makeWrapper $out/rtorrent $out/bin/rtorrent --set RTORRENT_CA_BUNDLE ${cacert}/etc/ssl/certs/ca-bundle.crt
+  '';
```

Comment:
There's most likely a more appropriate stage for this, just not sure which is the most suitable.